lib/tests/streamlit/config_test.py

```diff
@@ -2195,6 +2212,108 @@ def test_theme_inheritance_preserves_env_var_and_flag_precedence(self):
                     in config.get_where_defined("theme.linkColor").lower()
                 )
 
+    def test_theme_complex_inheritance_preserves_env_var_and_flag_precedence(self):
+        """Test that theme inheritance with sections/subsections preserves environment
+        variables and command line flags."""
+        theme_content = """
+        [theme]
+        base = "dark"
+        primaryColor = "#00ff00"
+
+        [theme.light]
+        font = "sans-serif"
+        backgroundColor = "#000000"
+
+        [theme.dark]
+        font = "serif"
+
+        [theme.sidebar]
+        textColor = "#ffffff"
+        borderColor = "#333333"
+
+        [theme.light.sidebar]
+        borderColor = "#999999"
+
+        [theme.dark.sidebar]
+        linkColor = "#cccccc"
+        """
+
+        with self._theme_file(theme_content) as theme_file:
+            # Config file references theme file and sets some overrides
+            config_toml = f"""
```

Comment:
If you have the following config:

```
[theme]
primaryColor = "#0"

[theme.dark]
primaryColor = "#1"

[theme.sidebar]
primaryColor = "#2"
```
 
What would be the expected color of `theme.dark.sidebar`?